README.md

```diff
@@ -1 +1,169 @@
+
+```haskell
+    it "is the inverse of daysAgo for just days" $
+      inverseFunctionsOnGen
+        daysAgoToDays
+        daysAgo
+        (((\d -> DaysAgo GT 0 d 0 0) <$> genValid) `suchThat` isValid)
+        (const [])
+```
+
+At this point I am confident that the `daysAgo` function works, but the `prettyRelativeDay` days does not lend itself to property testing so easily. So we will just write unit tests, one for every piece of expected output:
+
+```haskell
+    it "produces valid Strings" $ producesValidsOnValids prettyRelativeDay
+    let i da s = it (unwords ["Renders", show da, "as", show s]) $ prettyRelativeDay da `shouldBe` s
+    describe "renders these simple examples well" $ do
+      i (DaysAgo GT 5 0 0 0) "5 years ago"
+      i (DaysAgo GT 0 4 0 0) "4 months ago"
+      i (DaysAgo GT 0 0 3 0) "3 weeks ago"
+      i (DaysAgo GT 0 0 0 2) "2 days ago"
+      i (DaysAgo GT 0 0 0 1) "yesterday"
+      i (DaysAgo EQ 0 0 0 0) "today"
+      i (DaysAgo LT 0 0 0 1) "tomorrow"
+      i (DaysAgo LT 0 0 0 2) "in 2 days"
+      i (DaysAgo LT 0 0 3 0) "in 3 weeks"
+      i (DaysAgo LT 0 4 0 0) "in 4 months"
+      i (DaysAgo LT 5 0 0 0) "in 5 years"
+    describe "handles singular nouns well" $ do
+      i (DaysAgo GT 1 0 0 0) "1 year ago"
+      i (DaysAgo GT 0 1 0 0) "1 month ago"
+      i (DaysAgo GT 0 0 1 0) "1 week ago"
+      i (DaysAgo GT 0 0 0 1) "yesterday"
+      i (DaysAgo LT 0 0 0 1) "tomorrow"
+      i (DaysAgo LT 0 0 1 0) "in 1 week"
+      i (DaysAgo LT 0 1 0 0) "in 1 month"
+      i (DaysAgo LT 1 0 0 0) "in 1 year"
+```
+
+Note:
+
+- `"yesterday"`, `"today"` and `"tomorrow"` are tested separately.
+- Each level of granularity is tested.
+- Both singular and plural are tested for each of the levels of granularity.
+- For plurals, a different plural is used every time to make it less likely that there is a problem that only shows up for specific numbers.
+- There are no tests for more complicated values of DaysAgo where more than one part is nonzero. Frankly, that is because this library is not going to be mission-critical and I could not be bothered to go that far with the testing.
+
+## References
+
+The `pretty-relative-time` library is available [on Hackage](https://hackage.haskell.org/package/pretty-relative-time). Mergeful originated in the work on [Smos](https://smos.cs-syd.eu/), [Intray](https://intray.cs-syd.eu/) and [Tickler](https://tickler.cs-syd.eu/). This post is part of an effort to encourage contributions to [Smos](https://github.com/NorfairKing/smos). The simplest contribution could be to just try out smos and provide feedback on the experience. [Smos](https://github.com/NorfairKing/smos) is a purely functional semantic forest editor of a subset of YAML that is intended to replace Emacs' Org-mode for Getting Things Done.
```

Comment:
Note that if my assessment of turning the announcement post into an initial README is a good way forward, changing the last part that discusses Smos should perhaps be removed. I've not taken that liberty, but I'm making a note that maybe this relationship to Smos won't stand the test of time as well as the rest of the library documentation.